app/src/main/java/ckmah/mangoreader/UpdateService.java

```diff
@@ -0,0 +1,142 @@
+package ckmah.mangoreader;
+
+import android.app.IntentService;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Intent;
+import android.support.v7.app.NotificationCompat;
+import android.util.Log;
+
+import com.william.mangoreader.R;
+
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import ckmah.mangoreader.activity.MangoReaderActivity;
+import ckmah.mangoreader.model.MangaEdenMangaListItem;
+import ckmah.mangoreader.parse.MangaEden;
+
+public class UpdateService extends IntentService {
+    /**
+     * Creates an IntentService.  Invoked by your subclass's constructor.
+     * Used to name the worker thread, important only for debugging.
+     */
+    public UpdateService() {
+        super("UpdateService");
+    }
+
+    @Override
+    protected void onHandleIntent(Intent workIntent) {
+        try {
+            Log.d("UpdateService", "starting");
+            // Synchronously download list of all manga
+            MangaEden.MangaEdenList list = MangaEden
+                    .getMangaEdenServiceNoCache(this)
+                    .listAllManga()
+                    .execute()
+                    .body();
+
+            List<MangaEdenMangaListItem> updated = new ArrayList<>();
+
+            for (MangaEdenMangaListItem item : list.manga) {
```

Comment:
@williamxu That's an interesting idea -- it would probably speed up the time for the check to complete, but I don't think that time is currently a concern since it's running on a background thread that the user won't notice or think about until the reminder notification pops up. 

@ckmah "recently updated" in browse layout can be done with another sort on the all manga list by lastChapterDate, just like we're currently sorting it by hits for popularity.